mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+            error_msg += [
+                "Please set the CouchDB basic credentials via constructor args or `COUCH_CRED` variable"
+            ]
+        if not self.mcm_couchdb_lucene_url:
+            error_msg += [
+                "Please set the CouchDB Lucene connection url via constructor args or `MCM_LUCENE_URL` variable"
+            ]
+        if not self.mcm_application_url:
+            # Check if it is possible to form it from environment
+            # variables.
+            host_from_env: str = os.getenv("MCM_HOST", "")
+            port_from_env: str = os.getenv("MCM_PORT", "")
+            if host_from_env and port_from_env:
+                self.mcm_application_url = f"http://{host_from_env}:{port_from_env}/"
+            else:
+                error_msg += [
+                    (
+                        "Please set the McM application url via constructor args "
+                        "or set the enviroment variables `MCM_HOST` and `MCM_PORT`"
+                    )
+                ]
+
+        # Check and raise
+        if error_msg:
+            compress_msg = "\n".join(error_msg)
+            raise RuntimeError(compress_msg)
+
+        # Check the required mock users are available
+        # Otherwise, create them
+        if not mockup:
+            if not self.check_test_users():
+                self.__include_test_users()
+
+    def __http_request(
+        self,
+        base: str,
+        endpoint: str,
+        method: str,
+        data: dict = {},
+        headers: dict = {},
+    ) -> Response:
+        """
+        Executes a HTTP request.
+
+        Args:
+            base (str): Base URL, e.g: http://localhost:5984/
+            endpoint (str): Endpoint to consume, e.g: entity/document?p=1
+            method (str): HTTP method, e.g: POST
+            data (dict | None): Data to include in the request body.
+            headers (dict): Request headers
+        """
+        session = Session()
+        full_url: str = base + endpoint
+        req = Request(method=method, url=full_url, headers=headers, json=data)
+        prepared_req = session.prepare_request(req)
+        return session.send(prepared_req)
+
+    def couchdb_request(
+        self, endpoint: str, method: str, data: dict = {}, headers: dict = {}
+    ) -> Response:
+        """
+        Sends a HTTP request to CouchDB.
+
+        Args:
+            endpoint (str): Endpoint to consume, e.g: entity/document?p=1
+            method (str): HTTP method, e.g: POST
+            data (dict | None): Data to include in the request body.
+            headers (dict): Request headers
+        """
+        couchdb_headers = {"Authorization": self.mcm_couchdb_credential}
+        full_headers = {**headers, **couchdb_headers}
+        return self.__http_request(
+            base=self.mcm_couchdb_url,
+            endpoint=endpoint,
+            method=method,
+            data=data,
+            headers=full_headers,
+        )
+
+    def mcm_request(
```

Comment:
This would be a good use case for a lazily created `Session` with `headers=...` (one per role). It could then be used with all the normal `requests` methods.

Reply:
Also removed